FirebaseRemoteConfig/Tests/SwiftUnit/ConfigValueOrigTests.swift

```diff
@@ -0,0 +1,72 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+@testable import FirebaseRemoteConfig
+import XCTest
+
+class RemoteConfigValueOrigTests: XCTestCase {
+  func testConfigValueWithDifferentValueTypes() throws {
+    let valueA = "0.33333"
+    let dataA = try XCTUnwrap(valueA.data(using: .utf8))
+
+    let configValueA = RemoteConfigValue(data: dataA, source: .remote)
+    XCTAssertEqual(configValueA.stringValue, valueA)
+    XCTAssertEqual(configValueA.dataValue, dataA)
+    XCTAssertEqual(
+      configValueA.numberValue,
+      NSNumber(floatLiteral: 0.33333)
+    )
+    XCTAssertEqual(configValueA.boolValue, (valueA as NSString).boolValue)
+
+    let valueB = "NO"
+    let dataB = try XCTUnwrap(valueB.data(using: .utf8))
+    let configValueB = RemoteConfigValue(data: dataB, source: .remote)
+    XCTAssertEqual(configValueB.boolValue, (valueB as NSString).boolValue)
+
+    // Test JSON value.
+    let jsonDictionary = ["key1": "value1"]
+    let jsonArray = [["key1": "value1"], ["key2": "value2"]]
+
+    let jsonData = try JSONSerialization.data(withJSONObject: jsonDictionary, options: [])
+    let configValueC = RemoteConfigValue(data: jsonData, source: .remote)
+    XCTAssertEqual(configValueC.jsonValue as? [String: String], jsonDictionary)
```

Comment:
Should we have a more restrictive type on ConfigValue.jsonValue than `Any?`?

Reply:
Yes, but it will be tough to do. The comment for jsonValue is 
```
/// Gets a foundation object (NSDictionary / NSArray) by parsing the value as JSON.
```
but it could technically be a String or Int since those are also valid JSON. Going to merge this, but added to the Firebase 12 API doc for investigation.